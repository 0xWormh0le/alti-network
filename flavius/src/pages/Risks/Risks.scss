@import 'styles/_shared.scss';

.Risks {
  &__risks {
    margin: 20px $PAGE_MARGIN $PAGE_MARGIN;
    max-width: $MAX_CONTENT_WIDTH;
  }

  &__header {
    display: flex;
    padding: $LARGE_SPACING;
    margin-bottom: $LARGE_SPACING;
    background-color: $PALE_GREY_LIGHTEST;
    border-radius: 5px;
    gap: $SMALL_SPACING;

    &__filter {
      width: 16rem;
    }
  }

  .Table {
    &__table {
      padding-right: 1em;
    }

    &__heading {
      &-cell-sortable {
        padding: 0;
      }
    }

    &__row {
      &.ignored-risk {
        background-color: $PALE_GREY_LIGHTEST;

        &:hover {
          background-color: $PALE_GREY;
        }
      }

      .Table__cell:last-child {
        padding-right: 2em;
      }
    }
  }
}
