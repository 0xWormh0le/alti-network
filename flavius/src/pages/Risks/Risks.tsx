import React, { Fragment, useState, useMemo, useCallback, useEffect } from 'react'
import { ErrorBox, ModernTable } from '@altitudenetworks/component-library'
import PageTitle from 'components/elements/PageTitle'
import Filter, { FilterItem } from 'components/elements/Filter/Filter'
import { platformImages, platforms, riskTypeGroups } from 'config'
import EntityCountPagination from 'components/elements/EntityCountPagination'
import RiskIndicator from 'components/elements/RiskIndicator'
import RiskDescriptionCell from 'components/risks/RiskDescriptionCell'
import Button from 'components/elements/Button'
import FileCell from 'components/elements/FileCell'
import PersonCell from 'components/elements/PersonCell'
import DateAndTimeCell from 'components/elements/DateAndTimeCell'
import TableLoading from 'components/elements/TableLoading'
import Typography, { TypographyVariant } from 'components/elements/Typography'
import ActionsCell from 'components/risks/ActionsCell'
import {
  InformationalRiskTypeIds,
  SharingRiskTypeIds,
  RelationshipRiskTypeIds,
  ActivityRiskTypeIds
} from 'models/RiskCatalog'
import { resolvableRisk, mapper, handleSendEmail, alertableRisk, rowMapper, useCacheIncidentDates } from './riskUtils'
import { GENERAL_URLS } from 'api/endpoints'
import useRiskApiClient from 'api/clients/riskApiClient'
import { getCSVData } from 'util/csv'
import UI_STRINGS from 'util/ui-strings'
import { getFullPlatformIds } from 'util/platforms'
import { useQueryParam, useConfirmModal, useGetCachedPageSize } from 'util/hooks'
import { SensitiveContentStatus } from 'types/common'
import './Risks.scss'

const riskTypeGroupsForComponent = (
  riskTypeGroupsData: BasicRiskTypeGroup[],
  selected: boolean | ((data: BasicRiskTypeGroup) => boolean)
) =>
  riskTypeGroupsData.map((riskTypeGroup) => ({
    id: riskTypeGroup.groupType.toString(),
    value: riskTypeGroup.groupType.toString(),
    selected: typeof selected === 'function' ? selected(riskTypeGroup) : selected,
    renderComponent: (
      <div>
        <span style={{ marginLeft: '0.5rem' }}>{riskTypeGroup.name}</span>
      </div>
    )
  }))

const platformsForComponent = (
  platformsData: BasicPlatformData[],
  selected: boolean | ((data: BasicPlatformData) => boolean)
) =>
  platformsData.map((platform) => ({
    id: platform.platformId,
    value: platform.platformId,
    selected: typeof selected === 'function' ? selected(platform) : selected,
    renderComponent: (
      <div>
        <img src={platformImages[platform.platformId].Icon} alt={platform.platformName} />
        <span style={{ marginLeft: '0.5rem' }}>{platform.platformName}</span>
      </div>
    )
  }))

type StateFilterFunc = (selected: FilterUserVisibilityState) => FilterItem[]

const stateFilterData: StateFilterFunc = (selected) =>
  ['Active', 'Ignored', 'All'].map((label) => ({
    id: label.toLowerCase(),
    value: label.toLowerCase(),
    label,
    selected: label.toLowerCase() === selected
  }))

const defaultRiskTypeGroupFilterData = riskTypeGroupsForComponent(riskTypeGroups, true)

const defaultPlatformFilterData = platformsForComponent(platforms, true)

const initialPlatformIds = getFullPlatformIds()

const fields = ['Severity', 'Risk Type', 'Files At Risk', 'File Owner', 'Detected On']

// Translating between table and API
const orderingColumns = {
  'Detected On': 'datetime',
  Severity: 'severity'
}

const headerRow = [
  UI_STRINGS.CSV_HEADERS.SEVERITY,
  UI_STRINGS.CSV_HEADERS.RISK_TYPE,
  UI_STRINGS.CSV_HEADERS.FILES_AT_RISK,
  UI_STRINGS.CSV_HEADERS.LINK_INSPECTOR,
  UI_STRINGS.CSV_HEADERS.LINK_TO_FILE,
  UI_STRINGS.CSV_HEADERS.FILE_ID,
  UI_STRINGS.CSV_HEADERS.PLATFORM,
  UI_STRINGS.CSV_HEADERS.FILE_OWNER,
  UI_STRINGS.CSV_HEADERS.LINK_OWNERS_SPOTLIGHT,
  UI_STRINGS.CSV_HEADERS.DETECTED_ON
].join()

const riskTypes = ([] as number[])
  .concat(InformationalRiskTypeIds)
  .concat(SharingRiskTypeIds)
  .concat(RelationshipRiskTypeIds)
  .concat(ActivityRiskTypeIds)

const Risks: React.FC<{}> = () => {
  const [platformIds, setPlatformIds] = useQueryParam<string[]>('platformIds', initialPlatformIds, { pageNumber: 1 })
  const [riskTypeIds, setRiskTypeIds] = useQueryParam<number[]>('riskTypeIds', [], { pageNumber: 1 })
  const [pageNumber, setPageNumber] = useQueryParam<number>('pageNumber', 1)
  const [pageSize, setPageSize] = useGetCachedPageSize({ pageNumber: 1 })
  const [userVisibilityState, setUserVisibilityState] = useQueryParam<UserVisibilityState>(
    'userVisibilityState',
    'active',
    { pageNumber: 1 }
  )
  const [isExportingCsv, setIsExportingCsv] = useState(false)
  const [orderBy, setOrderBy] = useState('Detected On')
  const [sort, setSort] = useState<'asc' | 'desc'>('desc')
  const [isResolvingRisk, setIsResolvingRisk] = useState(false)
  const [displayRisksRs, setDisplayRisksRs] = useState(true) // same same risks data will be returned from useGetRisks and useUpdateRiskState, this flag is to determine whether to display the data from useGetRisks or not
  const showModal = useConfirmModal()

  const handleResetAllFilter = useCallback(() => {
    setPlatformIds(initialPlatformIds)
    setRiskTypeIds([])
    setUserVisibilityState('active')
  }, [setRiskTypeIds, setPlatformIds, setUserVisibilityState])

  const handleManagePermissions = useCallback((riskActionInfo: RiskAction) => {
    if (resolvableRisk(riskActionInfo)) {
      setIsResolvingRisk(false)
    }
  }, [])

  const { useGetRisks, useUpdateRiskState } = useRiskApiClient()

  const [risksRs, risksErr, isLoadingRisks] = useGetRisks({
    meta: {
      handleError: (err: any) => {
        const message: string = err.response.data.message
        throw new Error(message)
      },
      handleSuccess: () => {
        setDisplayRisksRs(true) // render the table with risksRs
      }
    },
    requestDetails: {
      pageNumber,
      pageSize,
      platformIds,
      riskTypeIds: riskTypeIds.map((x) => x.toString()),
      userVisibilityState,
      queryParams: {
        sort,
        orderBy: orderingColumns[orderBy]
      }
    }
  })
  const [updateRiskStateRs, updateRiskStateErr, isUpdatingRiskState, updateRiskState] = useUpdateRiskState({
    meta: {
      handleError: (err: any) => {
        const message: string = err.response.data.message
        throw new Error(message)
      },
      handleSuccess: () => {
        setDisplayRisksRs(false) // render the table with updateRiskStateRs
      }
    }
  })

  const cleanupIncidentDateCache = useCacheIncidentDates(risksRs)

  // Clean up cached incidents on dismount
  useEffect(() => {
    return () => {
      cleanupIncidentDateCache()
    }
  }, [cleanupIncidentDateCache])

  const displayRisks: DisplayRisk[] = risksRs?.risks.map(mapper) || []
  const risksData = displayRisksRs ? risksRs : updateRiskStateRs
  const hasError: boolean = Boolean(risksErr && !isLoadingRisks) || Boolean(updateRiskStateErr && !isUpdatingRiskState)
  const isLoadingTable: boolean = isLoadingRisks || isResolvingRisk || isUpdatingRiskState

  const riskTypeGroupsData = useMemo(() => {
    if (riskTypeIds.length) {
      const groups = riskTypeIds.map((riskTypeId) => Math.floor(riskTypeId / 1000))
      return riskTypeGroupsForComponent(riskTypeGroups, (data) => groups.includes(data.groupType))
    } else {
      return riskTypeGroupsForComponent(riskTypeGroups, true)
    }
  }, [riskTypeIds])

  const platformsData = useMemo(
    () => platformsForComponent(platforms, (data) => platformIds.includes(data.platformId)),
    [platformIds]
  )

  const onFilterPlatformChange = useCallback(
    (items: FilterItem[]) => {
      // update the platformIds in the query params
      setPlatformIds(items.filter((item) => item.selected).map((item) => item.value))
    },
    [setPlatformIds]
  )

  const onFilterRiskTypeGroupChange = useCallback(
    (items: FilterItem[]) => {
      // update the riskTypeIds in the query params
      const newRiskTypeGroups = items.filter((item) => item.selected).map((item) => item.value)

      if (newRiskTypeGroups.length === items.length) {
        setRiskTypeIds([])
      } else {
        setRiskTypeIds(riskTypes.filter((item) => newRiskTypeGroups.includes(Math.floor(item / 1000).toString())))
      }
    },
    [setRiskTypeIds]
  )

  const handleRiskStateChange = useCallback(
    (riskActionInfo: RiskAction, targetState: UserVisibilityState) => {
      const riskChangeArgs = {
        riskId: riskActionInfo.riskId,
        userVisibilityState: targetState,
        queryParams: {
          sort,
          orderBy: orderingColumns[orderBy],
          pageNumber,
          pageSize,
          platformIds,
          riskTypeIds: riskTypeIds.map((x) => x.toString()),
          userVisibilityState
        }
      }
      if (targetState === 'active') {
        showModal({
          message: (
            <Typography variant={TypographyVariant.H4}>{UI_STRINGS.RISKS.RISK_STATE_MODAL.ACTIVE_MESSAGE}</Typography>
          ),
          onConfirm: () => updateRiskState.call({ ...riskChangeArgs }),
          dialogTitle: UI_STRINGS.RISKS.RISK_STATE_MODAL.ACTIVE_TITLE,
          confirmButtonText: UI_STRINGS.BUTTON_LABELS.YES,
          cancelButtonText: UI_STRINGS.BUTTON_LABELS.CANCEL,
          confirmButtonActionType: 'primary',
          cancelButtonActionType: 'secondary'
        })
      } else {
        showModal({
          message: (
            <>
              <Typography variant={TypographyVariant.H4}>
                <p>{UI_STRINGS.RISKS.RISK_STATE_MODAL.IGNORE_MESSAGE}</p>
              </Typography>
              <p>{UI_STRINGS.RISKS.RISK_STATE_MODAL.IGNORE_DESC}</p>
            </>
          ),
          onConfirm: () => updateRiskState.call({ ...riskChangeArgs }),
          dialogTitle: UI_STRINGS.RISKS.RISK_STATE_MODAL.IGNORE_TITLE,
          confirmButtonText: UI_STRINGS.RISKS.RISK_STATE_MODAL.IGNORE_CONFIRM,
          cancelButtonText: UI_STRINGS.BUTTON_LABELS.CANCEL,
          confirmButtonActionType: 'alert',
          cancelButtonActionType: 'secondary'
        })
      }
    },
    [orderBy, pageNumber, pageSize, platformIds, userVisibilityState, riskTypeIds, showModal, sort, updateRiskState]
  )

  return (
    <div className='Risks'>
      <PageTitle title={UI_STRINGS.RISKS.TITLE} />

      <div className='Risks__risks'>
        <div className='Risks__header'>
          {platformsData && platformsData.length > 0 && (
            <Filter
              containerClass='Risks__header__filter'
              headerLabel='Platform'
              displayAllOption={true}
              tooltipText={UI_STRINGS.RISKS.TOOLTIP_FILTER}
              items={platformsData}
              resetItems={defaultPlatformFilterData}
              onSubmit={onFilterPlatformChange}
            />
          )}
          {riskTypeGroupsData && riskTypeGroupsData.length > 0 && (
            <Filter
              containerClass='Risks__header__filter'
              headerLabel='Risk Type'
              displayAllOption={true}
              tooltipText={UI_STRINGS.RISKS.TOOLTIP_FILTER}
              items={riskTypeGroupsData}
              resetItems={defaultRiskTypeGroupFilterData}
              onSubmit={onFilterRiskTypeGroupChange}
            />
          )}

          <Filter
            displayAllOption={false}
            headerLabel='State'
            items={stateFilterData(userVisibilityState)}
            isSingleSelect={true}
            onSubmit={(items) => setUserVisibilityState(items[0].value as UserVisibilityState)}
          />

          <Button action='simple' text={UI_STRINGS.BUTTON_LABELS.RESET} onClick={handleResetAllFilter} />
        </div>

        {hasError ? (
          risksErr?.error.includes('Only unconfigured') || updateRiskStateErr?.error.includes('Only unconfigured') ? (
            <ErrorBox secondaryMessage='' mainMessage={UI_STRINGS.ERROR_MESSAGES.NO_RESULTS} />
          ) : (
            <ErrorBox
              mainMessage={UI_STRINGS.ERROR_MESSAGES.SOMETHING_WRONG}
              secondaryMessage={UI_STRINGS.ERROR_MESSAGES.OUR_END}
            />
          )
        ) : (
          <Fragment>
            <ModernTable
              setSort={setSort}
              sort={sort}
              setOrderBy={setOrderBy}
              orderBy={orderBy}
              actions={(risk) => {
                const riskActionInfo: RiskAction = {
                  description: '',
                  email: risk.file.email,
                  file: risk.file,
                  platformId: risk.file.platformId,
                  plugin: {
                    id: risk.file.pluginId || '',
                    name: risk.file.pluginName || ''
                  },
                  userVisibilityState: risk.userVisibilityState,
                  riskId: risk.riskId,
                  riskTypeId: risk.riskTypeId,
                  webLink: risk.file.webLink,
                  owner: risk.owner
                }

                return (
                  <ActionsCell
                    onEmailAction={() => handleSendEmail(riskActionInfo)}
                    onManagePermissionsAction={() => handleManagePermissions(riskActionInfo)}
                    onIgnoreRiskAction={() => handleRiskStateChange(riskActionInfo, 'ignored')}
                    onPutBackToActiveAction={() => handleRiskStateChange(riskActionInfo, 'active')}
                    emailActionEnabled={alertableRisk(riskActionInfo)}
                    managePermissionsActionEnabled={resolvableRisk(riskActionInfo)}
                    onLockoutAction={() => null}
                    state={riskActionInfo.userVisibilityState as UserVisibilityState}
                    file={riskActionInfo.file}
                    riskId={riskActionInfo.riskId}
                    email={riskActionInfo.email}
                    riskTypeId={riskActionInfo.riskTypeId}
                    appId={riskActionInfo.plugin.id}
                    appName={riskActionInfo.plugin.name}
                    owner={riskActionInfo.owner}
                    platformId={riskActionInfo.platformId}
                  />
                )
              }}
              isLoading={isLoadingTable}
              loadingComponent={<TableLoading />}
              className='RisksTable'
              sortingHeaders={['Severity', 'Detected On']}
              items={displayRisks}
              fields={fields}
              scopedSlots={{
                Severity: (risk) => {
                  return <RiskIndicator value={risk.severity} />
                },
                'Risk Type': (risk) => {
                  return <RiskDescriptionCell displayRiskDescription={risk.description} />
                },
                'Files At Risk': (risk) => {
                  return (
                    <FileCell
                      sensitiveContentDetected={risk.sensitiveContent as SensitiveContentStatus}
                      fileResponse={risk.file}
                    />
                  )
                },
                'File Owner': (risk) => {
                  return <PersonCell personData={risk.owner} />
                },
                'Detected On': (risk) => {
                  return <DateAndTimeCell value={risk.datetime} />
                }
              }}
              rowClassName={(risk) => (risk.userVisibilityState === 'ignored' ? 'ignored-risk' : '')}
            />
            <EntityCountPagination
              onPageChange={setPageNumber}
              totalEntityCount={risksData?.riskCount || 0}
              entityCount={risksData?.risks?.length || 0}
              onExportCsv={() => {
                setIsExportingCsv(true)
                getCSVData(
                  {
                    baseUrl: `${GENERAL_URLS.RISKS}`,
                    dataKeyName: 'risks',
                    endpointName: 'risks',
                    pageSize: risksData?.pageSize || 0,
                    pageCount: risksData?.pageCount || 0,
                    queryParams: {
                      platformIds,
                      riskTypeIds,
                      userVisibilityState,
                      sort,
                      orderBy: orderingColumns[orderBy]
                    }
                  },
                  'Risks',
                  headerRow,
                  rowMapper
                ).then(() => setIsExportingCsv(false))
              }}
              isLoadingExportCsv={isExportingCsv}
              pageNumber={pageNumber}
              pageSize={pageSize}
              pageCount={risksData?.pageCount || 0}
              onPageSizeChange={setPageSize}
            />
          </Fragment>
        )}
      </div>
    </div>
  )
}

export default Risks
