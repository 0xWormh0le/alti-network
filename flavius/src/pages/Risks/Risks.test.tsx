import React from 'react'
import API from '@aws-amplify/api/lib'
import { renderWithRouter } from 'test/support/helpers'
import Risks from './Risks'
import { act } from 'react-dom/test-utils'
import { RenderResult } from '@testing-library/react'
import { risksResponse } from 'test/mocks'

jest.mock('@aws-amplify/api/lib')

let renderResult: RenderResult

describe('Risks', () => {
  const mock = risksResponse
  beforeEach(async () => {
    // @ts-ignore
    API.get.mockResolvedValue(mock)

    await act(async () => {
      renderResult = renderWithRouter(<Risks />)
      await new Promise((r) => setTimeout(r, 200))
    })
  })

  it('renders correctly', async () => {
    // @ts-ignore
    API.get.mockResolvedValue(mock)
    const { container } = renderResult
    expect(container).toMatchSnapshot()
  })

  it('has data items', async () => {
    // @ts-ignore
    API.get.mockResolvedValue(mock)
    const { container } = renderResult
    expect(container.querySelector('.Table__row')).toBeTruthy()
  })
})
