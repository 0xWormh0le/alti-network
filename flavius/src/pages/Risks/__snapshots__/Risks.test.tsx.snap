// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Risks renders correctly 1`] = `
<div>
  <div
    class="Risks"
  >
    <h1
      class="Typography Typography--h2 Typography--normal PageTitle"
    >
      Risks
    </h1>
    <div
      class="Risks__risks"
    >
      <div
        class="Risks__header"
      >
        <div
          class="Filter__container Risks__header__filter"
        >
          <div
            class="Filter__header"
          >
            <div
              class="Filter__header-label"
            >
              Platform
            </div>
            <div
              class="Filter__header-selected-text"
            >
              <span
                class="trigger"
              >
                All
              </span>
              <div
                class="Filter__header-arrow-down"
              />
            </div>
          </div>
          <div
            class="Filter__dropdown Filter__dropdown-close"
            style="max-height: 0px;"
          >
            <div
              class="Filter__dropdown-inner"
            >
              <div
                class="Filter__dropdown-item"
              >
                <input
                  checked=""
                  type="checkbox"
                />
                <div>
                  <img
                    alt="Google Workspace"
                    src="google-logo.svg"
                  />
                  <span
                    style="margin-left: 0.5rem;"
                  >
                    Google Workspace
                  </span>
                </div>
              </div>
              <div
                class="Filter__dropdown-item"
              >
                <input
                  checked=""
                  type="checkbox"
                />
                <div>
                  <img
                    alt="Microsoft 365"
                    src="office-365.svg"
                  />
                  <span
                    style="margin-left: 0.5rem;"
                  >
                    Microsoft 365
                  </span>
                </div>
              </div>
              <hr />
              <div
                class="Filter__dropdown-item"
              >
                <input
                  checked=""
                  type="checkbox"
                />
                All
              </div>
              <div
                class="Filter__action"
              >
                <button
                  class="Button Filter__action-apply Button--primary Button--disabled"
                  disabled=""
                  type="button"
                >
                  Apply
                </button>
                <button
                  class="Filter__action-reset"
                >
                  Reset filter
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          class="Filter__container Risks__header__filter"
        >
          <div
            class="Filter__header"
          >
            <div
              class="Filter__header-label"
            >
              Risk Type
            </div>
            <div
              class="Filter__header-selected-text"
            >
              <span
                class="trigger"
              >
                All
              </span>
              <div
                class="Filter__header-arrow-down"
              />
            </div>
          </div>
          <div
            class="Filter__dropdown Filter__dropdown-close"
            style="max-height: 0px;"
          >
            <div
              class="Filter__dropdown-inner"
            >
              <div
                class="Filter__dropdown-item"
              >
                <input
                  checked=""
                  type="checkbox"
                />
                <div>
                  <span
                    style="margin-left: 0.5rem;"
                  >
                    Sharing Risks
                  </span>
                </div>
              </div>
              <div
                class="Filter__dropdown-item"
              >
                <input
                  checked=""
                  type="checkbox"
                />
                <div>
                  <span
                    style="margin-left: 0.5rem;"
                  >
                    Relationship Risks
                  </span>
                </div>
              </div>
              <div
                class="Filter__dropdown-item"
              >
                <input
                  checked=""
                  type="checkbox"
                />
                <div>
                  <span
                    style="margin-left: 0.5rem;"
                  >
                    Activity Risks
                  </span>
                </div>
              </div>
              <div
                class="Filter__dropdown-item"
              >
                <input
                  checked=""
                  type="checkbox"
                />
                <div>
                  <span
                    style="margin-left: 0.5rem;"
                  >
                    Informational Risks
                  </span>
                </div>
              </div>
              <hr />
              <div
                class="Filter__dropdown-item"
              >
                <input
                  checked=""
                  type="checkbox"
                />
                All
              </div>
              <div
                class="Filter__action"
              >
                <button
                  class="Button Filter__action-apply Button--primary Button--disabled"
                  disabled=""
                  type="button"
                >
                  Apply
                </button>
                <button
                  class="Filter__action-reset"
                >
                  Reset filter
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          class="Filter__container"
        >
          <div
            class="Filter__header"
          >
            <div
              class="Filter__header-label"
            >
              State
            </div>
            <div
              class="Filter__header-selected-text"
            >
              <span>
                Active
              </span>
              <div
                class="Filter__header-arrow-down"
              />
            </div>
          </div>
          <div
            class="Filter__dropdown Filter__dropdown-close"
            style="max-height: 0px;"
          >
            <div
              class="Filter__dropdown-inner"
            >
              <div
                class="Filter__dropdown-item single"
              >
                Active
              </div>
              <div
                class="Filter__dropdown-item single"
              >
                Ignored
              </div>
              <div
                class="Filter__dropdown-item single"
              >
                All
              </div>
            </div>
          </div>
        </div>
        <button
          class="Button Button--simple"
          type="button"
        >
          Reset
        </button>
      </div>
      <div
        class="Table RisksTable"
      >
        <table
          class="Table__table"
        >
          <thead>
            <tr
              class="Table__heading"
            >
              <th
                class="Table__heading-cell-sortable"
              >
                <span>
                  Severity
                </span>
                <span
                  class="Table__heading-sorting-state"
                >
                  <svg
                    class=""
                    height="11"
                    viewBox="0 0 7 11"
                    width="7"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M3.884 10.71l-.016.011-.051.045a.492.492 0 0 1-.636-.056L.355 7.884A.504.504 0 0 1 .3 7.246l.057-.067a.493.493 0 0 1 .705-.003l2.471 2.472L6.01 7.172a.504.504 0 0 1 .637-.055l.068.056a.493.493 0 0 1 0 .705l-2.83 2.831zM3.884.276L3.868.264 3.817.219a.492.492 0 0 0-.636.056L.355 3.102a.504.504 0 0 0-.055.637l.057.068a.493.493 0 0 0 .705.002l2.471-2.472L6.01 3.813a.504.504 0 0 0 .637.056l.068-.057a.493.493 0 0 0 0-.704L3.886.276z"
                      fill="currentColor"
                      fill-rule="evenodd"
                    />
                  </svg>
                </span>
              </th>
              <th
                class="Table__heading-cell"
              >
                <span>
                  Risk Type
                </span>
              </th>
              <th
                class="Table__heading-cell"
              >
                <span>
                  Files At Risk
                </span>
              </th>
              <th
                class="Table__heading-cell"
              >
                <span>
                  File Owner
                </span>
              </th>
              <th
                class="Table__heading-cell-sortable"
              >
                <span>
                  Detected On
                </span>
                <span
                  class="Table__heading-sorting-state"
                >
                  â†“
                </span>
              </th>
              <th
                class="Table_heading-cell"
              />
            </tr>
          </thead>
          <tbody>
            <tr
              class="Table__row"
            >
              <td
                class="Table__cell"
              >
                <div
                  class="RiskIndicator RiskIndicator--bar"
                >
                  <div
                    class="RiskIndicator__inner RiskIndicator--medium"
                  />
                </div>
              </td>
              <td
                class="Table__cell"
              >
                <div
                  class="RiskDescriptionCell"
                >
                  <span>
                    File Shared by Link Externally - Company Owned
                  </span>
                </div>
              </td>
              <td
                class="Table__cell"
              >
                <div
                  class="FileCell FileCell__multiline"
                >
                  <div
                    class="FileCell__icons"
                  >
                    <span
                      class="AppIndicator"
                    >
                      <div
                        class="platform-container"
                      >
                        <a
                          class="trigger"
                          href="webLink"
                          rel="noopener noreferrer"
                          target="_blank"
                        >
                          <div>
                            <div>
                              <div
                                class="platform-icon-container"
                                data-testid="platform-icon"
                              >
                                <img
                                  alt="Google Workspace"
                                  class="platform-icon"
                                  src="google-logo.svg"
                                  title="Google Workspace"
                                />
                              </div>
                            </div>
                          </div>
                        </a>
                      </div>
                    </span>
                    <div
                      style="margin-right: 0.3em;"
                    />
                    <div
                      class="file_icon_container"
                    >
                      <div
                        class="trigger"
                      >
                        <img
                          alt="mimeType"
                          class="file_icon"
                          src="icon-small-file.png"
                        />
                      </div>
                    </div>
                  </div>
                  <a
                    class="FileCell__link"
                    data-testid="file-cell-link"
                    href="/dashboard/file/1_4sZI4rBgwnC2VDUKCSLFHMJc77oWEI2?owner=bobbie@thoughtlabs.io&selection=sensitive&platformId=gsuite&sensitive=true"
                  >
                    <div>
                      <div>
                        ptinpoly.c
                      </div>
                      <div
                        class="FileCell__spacer-top-5"
                      >
                        <div
                          class="sensitiveFileAlert trigger"
                          data-testid="sensitiveFileAlert"
                        >
                          <span
                            class="Typography Typography--body Marker Marker__sensitive_content"
                          >
                            Sensitive Content
                          </span>
                        </div>
                      </div>
                    </div>
                  </a>
                </div>
              </td>
              <td
                class="Table__cell"
              >
                <div
                  class="PersonCell"
                >
                  <a
                    class="PersonCell__link"
                    href="/dashboard/spotlight/bobbie%40thoughtlabs.io"
                  >
                    <div
                      class="PersonCell__email"
                    >
                      <span
                        class="trigger"
                      >
                        bobbie@thoughtlab...
                      </span>
                      <span
                        class="Typography Typography--body PersonCell__email-ext trigger Marker Marker__ext"
                      >
                        EXT
                      </span>
                    </div>
                  </a>
                </div>
              </td>
              <td
                class="Table__cell"
              >
                <div
                  class="DateAndTimeCell"
                >
                  <time
                    class="DateAndTimeCell__date"
                    datetime="1566244504000"
                  >
                    Aug 19, 2019
                  </time>
                  <div
                    class="DateAndTimeCell__time-container"
                  >
                    <time
                      class="DateAndTimeCell__time"
                      datetime="1566244504000"
                    >
                      12:55pm
                    </time>
                  </div>
                </div>
              </td>
              <td
                class="Table__cell"
              >
                <div
                  class="ActionsCell"
                >
                  <div
                    class="DropdownWrapper"
                    tabindex="1"
                  >
                    <button
                      class="Button Button--dropdown Button--size-medium Button--secondary"
                      type="button"
                    >
                      Actions
                    </button>
                    <svg
                      class="DropdownPopup--arrow-off"
                      height="8px"
                      version="1.1"
                      viewBox="0 0 12 8"
                      width="12px"
                    >
                      <polygon
                        id="path-1"
                        points="6 8 12 0 0 0"
                      />
                    </svg>
                    <ul
                      class="DropdownPopup"
                    >
                      <span
                        class="DropdownPopup--arrow"
                      />
                      <li
                        class="DropdownPopup--item"
                      >
                        <span>
                          Alert File Owner
                        </span>
                      </li>
                      <li
                        class="DropdownPopup--item"
                      >
                        <a
                          href="/dashboard/resolve/8c488709dbae483aa0a3576b6fd6589e?riskId=8c488709dbae483aa0a3576b6fd6589e&riskTypeId=1020&email=risk_creator%40thoughtlabs.io&appName=name&appId=id&fileCount=1&platformId=gsuite"
                        >
                          Resolve Risk
                        </a>
                      </li>
                      
                      <li
                        class="DropdownPopup--item"
                      >
                        <span>
                          Ignore Risk
                        </span>
                      </li>
                      
                    </ul>
                  </div>
                </div>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        class="EntityCountPagination"
      >
        <div>
          <button
            class="Button EntityCountPagination__export-button Button--secondary"
            type="button"
          >
            Export All
          </button>
        </div>
        <div
          class="EntityCountPagination__pagination-info"
        >
          <div
            class="EntityCountPagination__page-size EntityCountPagination__items-count"
          >
            <span>
              Displaying
            </span>
            <div
              class="EntityCountPagination__controller"
            >
              <select
                class="FormControl FormControl--flat"
              >
                <option
                  class="FormControl FormControl--flat"
                  value="10"
                >
                  10
                </option>
                <option
                  class="FormControl FormControl--flat"
                  value="20"
                >
                  20
                </option>
                <option
                  class="FormControl FormControl--flat"
                  value="25"
                >
                  25
                </option>
                <option
                  class="FormControl FormControl--flat"
                  value="50"
                >
                  50
                </option>
              </select>
            </div>
            <span>
              Items Per Page
            </span>
          </div>
          <div
            class="Pagination"
          >
            <div
              class="Pagination__group"
            >
              <div
                class="PaginationItem PaginationItem--side"
              >
                <button
                  class="Typography Typography--body-tiny PaginationItem__link PaginationItem__link--disabled"
                >
                  1
                </button>
              </div>
            </div>
            <div
              class="Pagination__group"
            >
              <div
                aria-label="prev"
                class="PaginationItem PaginationItem--arrow"
              >
                <button
                  class="Typography Typography--body-tiny PaginationItem__link PaginationItem__link--disabled"
                >
                  <svg
                    class="Pagination__arrow"
                    height="20"
                    viewBox="0 0 20 20"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <polygon
                      fill="currentColor"
                      points="5,10 15,1 15,19"
                    />
                  </svg>
                </button>
              </div>
              <div
                aria-label="1"
                class="PaginationItem PaginationItem--number"
              >
                <button
                  class="Typography Typography--body-tiny PaginationItem__link PaginationItem__link--active"
                >
                  1
                </button>
              </div>
              <div
                aria-label="2"
                class="PaginationItem PaginationItem--number"
              >
                <button
                  class="Typography Typography--body-tiny PaginationItem__link"
                >
                  2
                </button>
              </div>
              <div
                aria-label="3"
                class="PaginationItem PaginationItem--number"
              >
                <button
                  class="Typography Typography--body-tiny PaginationItem__link"
                >
                  3
                </button>
              </div>
              <div
                aria-label="4"
                class="PaginationItem PaginationItem--number"
              >
                <button
                  class="Typography Typography--body-tiny PaginationItem__link"
                >
                  4
                </button>
              </div>
              <div
                aria-label="5"
                class="PaginationItem PaginationItem--number"
              >
                <button
                  class="Typography Typography--body-tiny PaginationItem__link"
                >
                  5
                </button>
              </div>
              <div
                aria-label="next"
                class="PaginationItem PaginationItem--arrow"
              >
                <button
                  class="Typography Typography--body-tiny PaginationItem__link"
                >
                  <svg
                    class="Pagination__arrow"
                    height="20"
                    viewBox="0 0 20 20"
                    width="20"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <polygon
                      fill="currentColor"
                      points="15,10 5,1 5,19"
                    />
                  </svg>
                </button>
              </div>
            </div>
            <div
              class="Pagination__group"
            >
              <div
                class="PaginationItem PaginationItem--side"
              >
                <button
                  class="Typography Typography--body-tiny PaginationItem__link"
                >
                  49
                </button>
              </div>
            </div>
            <form
              class="PageNumberInput"
            >
              <div
                class="Typography Typography--body-tiny Typography--semibold PageNumberInput__label"
              >
                Go to page
              </div>
              <div
                class="PageNumberInput__input-wrapper"
              >
                <input
                  class="PageNumberInput__input"
                  type="text"
                  value=""
                />
              </div>
              <button
                class="Button PageNumberInput__submit Button--pagination Button--disabled"
                disabled=""
                type="submit"
              >
                GO
              </button>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
